### DISCUSSION TITLE
rule to exclude alerts from whitelisted ip's

### DISCUSSION THREAD

**Discussion by Threepwood77** [wazuh/wazuh#15635] [Q&A] [answered]
Hello,
I have implemented the Wazuh 4.13 suite on an Ubutun server and it works very well.
I need help to keep false positives out of my alerts, as they are coming from known IP addresses and therefore to be whitelisted.

What I did, I added the two IP addresses to whitelist in a :

> /var/ossec/etc/lists/myaddressip.cdb

Then I declared this list in the following file:

> /var/ossec/etc/ossec.conf

This way (I added a line, the last):

>     
    <decoder_dir>ruleset/decoders</decoder_dir>
    <rule_dir>ruleset/rules</rule_dir>
    <rule_exclude>0215-policy_rules.xml</rule_exclude>
    <list>etc/lists/audit-keys</list>
    <list>etc/lists/amazon/aws-eventnames</list>
    <list>etc/lists/security-eventchannel</list>
    <list>etc/lists/myaddressip</list>

Then I added a rule to the one I was interested in, in this case: **31533**
So in the : 

> /var/ossec/ruleset/rules/0270-web_appsec_rules.xml

The original rule is as follows:

`<rule id="31533" level="10" timeframe="20" frequency="8">
<if_matched_sid>31530</if_matched_sid>
<same_source_ip />
<description>High amount of POST requests in a small period of time (likely bot).</description>
<group>pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
<mitre>
<id>T1498</id>
</mitre>
</rule>
`

I have added a modified copy just below:

`<rule id="100000" level="10" timeframe="20" frequency="8">
<if_matched_sid>31530</if_matched_sid>
<same_source_ip />
<list field="srcip" lookup="address_match_key">etc/lists/myaddressip</list>
<description>High amount of POST requests in a small period of time (likely bot). From whitelisted IP</description>
<group>pci_dss_6.5,pci_dss_11.4,gdpr_IV_35.7.d,nist_800_53_SA.11,nist_800_53_SI.4,tsc_CC6.6,tsc_CC7.1,tsc_CC8.1,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
<mitre>
<id>T1498</id>
</mitre>
</rule>
`

I restarted wazuh-manager, I didn't get any errors.
But I keep getting the usual alerts when I should have at least this description if the ip is listed: 

High amount of POST requests in a small period of time (likely bot). **From whitelisted IP**

Which is not the case.
I think I didn't do the right thing but I must admit I'm a bit dry here...
If anyone can help me 😊
Thanks in advance.

**Comment by s-ocando** [comment, marked as answer]
Hi @Threepwood77, 

The best way to create an exception to a rule is to create a child rule with the given condition. 

The Wazuh analysis engine processes each event by determining whether it fulfills the matching criteria of each rule one by one, and once it matches a rule it repeats the process only taking into account the children of that rule. This process is repeated until a matching rule does not have any matching children. The last rule to match is the one that determines the ID, level and description of the alert.

As such, your custom rule should not be a copy of the original rule but a child to it, for example:
```
<rule id="100003" level="3">
  <if_sid>31533</if_sid>
  <list field="srcip" lookup="address_match_key">etc/lists/myaddressip</list>
  <description>High amount of POST requests in a small period of time (likely bot) from list of allowed IPs</description>
</rule>
```

Store this custom rule under ``/var/ossec/etc/rules/local_rules.xml``, otherwise it will be overwritten during upgrades. Remember to restart the Wazuh manager after editing the rules. 

If you want to see more details of the analysis process,  you can test your logs with `/var/ossec/bin/wazuh-logtest` and use the verbose option `-v`.

```
**Rule debugging:
	Trying rule: 4 - Generic template for all web rules.
		*Rule 4 matched
		*Trying child rules
	Trying rule: 31100 - Access log messages grouped.
		*Rule 31100 matched
		*Trying child rules
	Trying rule: 31108 - Ignored URLs (simple queries).
	Trying rule: 31511 - Blacklisted user agent (wget).
	Trying rule: 31115 - URL too long. Higher than allowed on most browsers. Possible attack.
	Trying rule: 31103 - SQL injection attempt.
	Trying rule: 31104 - Common web attack.
	Trying rule: 31105 - XSS (Cross Site Scripting) attempt.
	Trying rule: 31110 - PHP CGI-bin vulnerability attempt.
	Trying rule: 31109 - MSSQL Injection attempt (/ur.php, urchin.js)
	Trying rule: 31164 - SQL injection attempt.
	Trying rule: 31165 - SQL injection attempt.
	Trying rule: 31170 - SQL injection attempt.
	Trying rule: 31171 - SQL injection attempt.
	Trying rule: 31501 - WordPress Comment Spam (coming from a fake search engine UA).
	Trying rule: 31502 - TimThumb vulnerability exploit attempt.
	Trying rule: 31503 - osCommerce login.php bypass attempt.
	Trying rule: 31504 - osCommerce file manager login.php bypass attempt.
	Trying rule: 31505 - TimThumb backdoor access attempt.
	Trying rule: 31506 - Cart.php directory transversal attempt.
	Trying rule: 31507 - MSSQL Injection attempt (ur.php, urchin.js).
	Trying rule: 31508 - Blacklisted user agent (known malicious user agent).
	Trying rule: 31512 - Uploadify vulnerability exploit attempt.
	Trying rule: 31513 - BBS delete.php exploit attempt.
	Trying rule: 31514 - Simple shell.php command execution.
	Trying rule: 31515 - PHPMyAdmin scans (looking for setup.php).
	Trying rule: 31516 - Suspicious URL access.
	Trying rule: 31550 - Anomaly URL query (attempting to pass null termination).
	Trying rule: 31101 - Web server 400 error code.
	Trying rule: 31120 - Web server 500 error code (server error).
	Trying rule: 31530 - POST request received.
		*Rule 31530 matched
		*Trying child rules
	Trying rule: 31531 - Ignoring often post requests inside /wp-admin and /admin.
	Trying rule: 31533 - High amount of POST requests in a small period of time (likely bot).
	
**Phase 3: Completed filtering (rules).
	id: '31530'
	level: '3'
	description: 'POST request received.'
	groups: '['web', 'appsec', 'attack']'
	firedtimes: '1'
	mail: 'False'	
```	



Let us know if you have any questions.

**Comment by Threepwood77** [comment]
Hello , 
thank you for your answer and your help 

I deleted my old entries and added a new rule in the following file:

> /var/ossec/etc/rules/local_rules.xml

the rule:

`  <rule id="100002" level="10">
           <if_sid>31533</if_sid>
            <list field="srcip" lookup="address_match_key">etc/lists/iptwins</list>
             <description>High amount of POST requests in a small period of time (likely bot) from list of allowed IPs</description>
     </rule>`

I put "level 10" because at first I wanted to test the setting and continue to see the alerts but with a different description.

But it doesn't seem to work, I still get the same alerts, from addresses that are in my whitelist and the description is the original one.

I wanted to use your example to test the rule, but it doesn't work for me, it waits for me to type something, so I set the rule to 100002
everything scrolls and then remains blocked (I imagine until the next alert in the logs), except that even in the meantime, I still receive the original alerts as indicated above.

I'm not sure if I'm using this command correctly ?

Thanks for your help.


any idea ?

Thank you !

**Reply by s-ocando** [reply]
Hello,

To use the wazuh-logtest, enter a log and see what rule it triggers. As rule 31533 triggers if 8 POST requests are received within 20 seconds,  you should enter the log that triggers rule 31530 - "POST request received" eight times.  See the following GIF.

![logtest](https://user-images.githubusercontent.com/61882981/211770235-2d8acb6e-ed03-4700-aa32-9f9d442744bb.gif)



Note that for each log analyzed, the `firedtimes` counter on "Phase 3: Completed filtering (rules)" increases. After 8 POST requests, our custom rule 100003 "High amount of POST requests in a small period of time (likely bot) from list of allowed IPs" triggers as the `scrip` in the log, 10.11.12.13, is included in my CDB list. 

Alternatively, you can use the Wazuh logtest from the user interface and paste eight logs. The difference in this case is that verbose mode is not available, hence the suggestion to use the console to view the analysis process. See attached image.

![image](https://user-images.githubusercontent.com/61882981/211766856-726c904a-0c71-4cd4-9da6-c4cfb8cd90d6.png)

If your custom rule still doesn't work, share some of your POST request logs and your CDB list to help you debug. Make sure to suppress sensitive information.
